pkg/gohai/filesystem/filesystem_nix.go

```diff
@@ -0,0 +1,189 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build darwin || linux
+// +build darwin linux
+
+package filesystem
+
+import (
+	"fmt"
+	"strings"
+	"syscall"
+
+	log "github.com/cihub/seelog"
+	"github.com/moby/sys/mountinfo"
+)
+
+var getFSSizeFn = getFSSize
+var getFSDevFn = getFSDev
+
+// These FS types should be excluded from listing
+var IgnoredFSTypes = map[string]struct{}{
+	"autofs":      {},
+	"debugfs":     {},
+	"devfs":       {},
+	"devpts":      {},
+	"devtmpfs":    {},
+	"fuse.portal": {},
+	"fusectl":     {},
+	"ignore":      {},
+	"kernfs":      {},
+	"mqueue":      {},
+	"none":        {},
+	"proc":        {},
+	"rpc_pipefs":  {},
+	"squashfs":    {},
+	"subfs":       {},
+	"sysfs":       {},
+}
+
+// These FS types are known to be remote
+var RemoteFSTypes = map[string]struct{}{
+	"acfs":       {},
+	"afs":        {},
+	"auristorfs": {},
+	"coda":       {},
+	"fhgfs":      {},
+	"gpfs":       {},
+	"ibrix":      {},
+	"ocfs2":      {},
+	"vxfs":       {},
+}
+
+// returns the size of the filesystem
+func getFSSize(mount *mountinfo.Info) (uint64, error) {
+	var statfs syscall.Statfs_t
+	if err := syscall.Statfs(mount.Mountpoint, &statfs); err != nil {
+		return 0, fmt.Errorf("statfs %s: %v", mount.Source, err)
+	}
+
+	sizeKB := statfs.Blocks * uint64(statfs.Bsize) / 1024
+	return sizeKB, nil
+}
+
+// returns the dev id of the given filesystem
+// the return type is interface{} because `syscall.Stat_t` uses different types for Dev depending on the platform
+func getFSDev(mount *mountinfo.Info) (interface{}, error) {
+	var stat syscall.Stat_t
+	if err := syscall.Stat(mount.Mountpoint, &stat); err != nil {
+		return 0, fmt.Errorf("stat %s: %w", mount.Source, err)
+	}
+
+	return stat.Dev, nil
+}
+
+func getFileSystemInfo() ([]MountInfo, error) {
+	return getFileSystemInfoWithMounts(nil)
+}
+
+// returns whether to use the new mountInfo instead of the old one
+func replaceDev(old, new MountInfo) bool {
```

Comment:
See https://github.com/coreutils/coreutils/blob/master/src/df.c#L775-L797 for the implementation in `df`